satpy/composites/__init__.py

```diff
@@ -770,60 +770,104 @@ class DayNightCompositor(GenericCompositor):
+        Args:
+            lim_low (float): lower limit of Sun zenith angle for the
+                             blending of the given channels
+            lim_high (float): upper limit of Sun zenith angle for the
+                             blending of the given channels
+        """
+        self.lim_low = lim_low
+        self.lim_high = lim_high
+        super(DayNightCompositor, self).__init__(*args, **kwargs)
+
+    def __call__(self, projectables, *args, **kwargs):
+
+        day_data = projectables[0]
+        night_data = projectables[1]
+
+        lim_low = np.cos(np.deg2rad(self.lim_low))
+        lim_high = np.cos(np.deg2rad(self.lim_high))
+        try:
+            coszen = xu.cos(xu.deg2rad(projectables[2]))
+        except IndexError:
+            from pyorbital.astronomy import cos_zen
+            LOG.debug("Computing sun zenith angles.")
+            lons, lats = day_data.attrs["area"].get_lonlats_dask(CHUNK_SIZE)
+            coszen = xr.DataArray(cos_zen(day_data.attrs["start_time"],
+                                          lons, lats),
+                                  dims=['y', 'x'],
+                                  coords=[day_data['y'], day_data['x']])
+        # Calculate blending weights
+        coszen -= np.min((lim_high, lim_low))
+        coszen /= np.abs(lim_low - lim_high)
+        coszen = coszen.clip(0, 1)
+
+        # Apply enhancements to get images
+        day_data = enhance2dataset(day_data)
+        night_data = enhance2dataset(night_data)
+
+        # Adjust bands so that they match
+        # L/RGB -> RGB/RGB
+        # LA/RGB -> RGBA/RGBA
+        # RGB/RGBA -> RGBA/RGBA
+        day_data = add_bands(day_data, night_data['bands'])
+        night_data = add_bands(night_data, day_data['bands'])
+
+        # Get merged metadata
+        attrs = combine_metadata(day_data, night_data)
+
+        # Blend the two images together
+        data = (1 - coszen) * night_data + coszen * day_data
+        data.attrs = attrs
+
+        # Split to separate bands so the mode is correct
+        data = [data.sel(bands=b) for b in data.bands]
+
+        res = super(DayNightCompositor, self).__call__(data, **kwargs)
 
         return res
 
 
+def enhance2dataset(dset):
+    """Apply enhancements to dataset *dset* and return the resulting data
+    array of the image."""
+    attrs = dset.attrs
+    img = get_enhanced_image(dset)
+    # Clip image data to interval [0.0, 1.0] and replace nan values
+    data = img.data.clip(0.0, 1.0).fillna(0.0)
+    data.attrs = attrs
+
+    return data
+
+
+def add_bands(data, bands):
+    """Add bands so that they match *bands*"""
+    # Add R, G and B bands, remove L band
+    if 'L' in data.bands.data and 'R' in bands.data:
+        lum = data.sel(bands='L')
+        new_data = xr.concat((lum, lum, lum), dim='bands')
+        new_data['bands'] = ['R', 'G', 'B']
+        data = new_data
+    # Add alpha band
+    if 'A' not in data.bands.data and 'A' in bands.data:
```

Comment:
Could you user `data['bands']` instead ?

Reply:
Will do!